proto/cosmos/auth/v1beta1/auth.proto

```diff
@@ -18,27 +18,34 @@ message BaseAccount {
 
   option (cosmos_proto.implements_interface) = "cosmos.auth.v1beta1.AccountI";
 
-  string              address = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
-  google.protobuf.Any pub_key = 2 [(gogoproto.jsontag) = "public_key,omitempty", (amino.field_name) = "public_key"];
+  string              address = 1 [
+    (cosmos_proto.scalar) = "cosmos.AddressString"];
+
+  google.protobuf.Any pub_key = 2 [
+    (gogoproto.jsontag) = "public_key,omitempty",
+    (amino.field_name) = "public_key"];
+
   uint64              account_number = 3;
   uint64              sequence       = 4;
 }
 
 // ModuleAccount defines an account for modules that holds coins on a pool.
 message ModuleAccount {
   option (amino.name)                        = "cosmos-sdk/ModuleAccount";
+  option (amino.message_encoding)            = "module_account";
```

Comment:
The legacy encoder features a custom encoding for `ModuleAccount`
https://github.com/cosmos/cosmos-sdk/blob/41a3dfeced2953beba3a7d11ec798d17ee19f506/x/auth/types/account.go#L240-L254.

The protoreflect encoder provides this by registering message encoding overrides by key.  There this encoding corresponds to this registration:
https://github.com/cosmos/cosmos-sdk/blob/732bcafde4b280d8516ddc8abaad2ad04f0ddb58/x/tx/aminojson/json_marshal.go#L41
and this implementation:
https://github.com/cosmos/cosmos-sdk/blob/583181cce8f2ea6ab33efedecb19168474f0e6db/x/tx/aminojson/encoder.go#L101-L124